sample-app/src/main/java/com/joinforage/android/example/pos/receipts/CPayPrinter.kt

```diff
@@ -0,0 +1,120 @@
+package com.joinforage.android.example.pos.receipts
+
+import android.os.Bundle
+import com.pos.sdk.printer.PrinterDevice
+import com.pos.sdk.printer.param.MultipleTextPrintItemParam
+import com.pos.sdk.printer.param.PrintItemAlign
+import com.pos.sdk.printer.param.TextPrintItemParam
+
+// TODO: write tests for the classes and methods in this file
+//  as they are business logic and are straightforward to test
+
+/**
+ * This class and the methods it uses effectively map
+ * ReceiptLayouts, ReceiptLayoutLines, and ReceiptLineParts
+ * to the data structures that the CPay SDK uses for printing
+ */
+internal class CPayPrinter(private val cpayPrinter: PrinterDevice) {
+    fun setLayout(layout: ReceiptLayout) {
+        // clear the buffer before setting anything new
+        cpayPrinter.clearBufferArea()
+
+        // transform and write lines to the CPay printer's buffer
+        layout.lines.forEach {
+            CPayLine.of(it).addLineToPrinter(cpayPrinter)
+        }
+    }
+    fun print() {
+        val state = cpayPrinter.printSync(Bundle())
+        println("result is ${state.stateCode},msg is ${state.stateMsg}")
```

Comment:
do we need this `println`

Reply:
https://github.com/teamforage/forage-android-sdk/pull/148/commits/39149b28e4d57c93fab9d3ea3c7c17a7e5b2b098